### ISSUE TITLE
Add tag to configure UnusedImport star symbol [Java][BestPractices]

### ISSUE THREAD

**Issue by schmittjoaopedro** [pmd/pmd#2877]
I like to use the `UnusedImport` rule to prevent developers from forget to delete imports not used anymore. However, I had to ignore this rule in my projects because I don't want to add every single import for a bunch of dependencies that a class have, I mean the use of the start `*` symbol. Would be nice if the `UnusedImports` rule could be configured to decide when I want to consider the star symbol or not.

**Comment by adangel** [comment]
I'm not sure, I understand the issue. The UnusedImport rule should automatically consider on-demand imports. If no single class is used, then the import is unused. If only one class is used, then the rule should not trigger.

How do you execute PMD? command line, build plugin, ... ? It might be related to a incomplete auxclasspath for typeresolution: We can only correctly detect on-demand imports, if we can resolve the referenced types...

**Comment by schmittjoaopedro** [comment]
Sorry, let me clarify.

I would like to have the possibility to ignore PMD failing when detecting on-demand imports (star symbol), ex: 
```java
import java.io.File;  // not referenced or required
import java.util.*;   // <- it would nice to be able to disable PMD failing for these cases.

public class Foo {}
```

I think in some configuration property like this:
```xml
<rule ref="category/java/bestpractices.xml/UnusedImports" />
    <properties>
        <property name="ignoreOnDemandImports" value="false" />
    </properties>
</rule>
```

**Comment by oowekyala** [comment]
Do I understand correctly, that in the example you provided, you would like no violation for `java.util.*` but one for `java.io.file`?

**Comment by schmittjoaopedro** [comment]
yes, that's it.
